tooling/nargo_cli/src/cli/fuzz_cmd.rs

```diff
@@ -0,0 +1,489 @@
+        )?;
+    }
+
+    Ok(fuzzing_reports)
+}
+
+#[allow(clippy::too_many_arguments)]
+fn run_fuzzing_harness<S: BlackBoxFunctionSolver<FieldElement> + Default>(
+    file_manager: &FileManager,
+    parsed_files: &ParsedFiles,
+    package: &Package,
+    fn_name: &str,
+    show_output: bool,
+    foreign_call_resolver_url: Option<&str>,
+    root_path: Option<PathBuf>,
+    package_name: Option<String>,
+    compile_options: &CompileOptions,
+    fuzz_folder_config: &FuzzFolderConfig,
+    fuzz_execution_config: &FuzzExecutionConfig,
+) -> FuzzingRunStatus {
+    // This is really hacky but we can't share `Context` or `S` across threads.
+    // We then need to construct a separate copy for each test.
+
+    let (mut context, crate_id) = prepare_package(file_manager, parsed_files, package);
+    check_crate(&mut context, crate_id, compile_options)
+        .expect("Any errors should have occurred when collecting fuzzing harnesses");
+
+    let pattern = FunctionNameMatch::Exact(vec![fn_name.to_string()]);
+    let fuzzing_harnesses =
+        context.get_all_fuzzing_harnesses_in_crate_matching(&crate_id, &pattern);
+    let (_, fuzzing_harness) = fuzzing_harnesses.first().expect("Fuzzing harness should exist");
+
+    nargo::ops::run_fuzzing_harness::<S>(
+        &mut context,
+        fuzzing_harness,
+        show_output,
+        foreign_call_resolver_url,
+        root_path,
+        package_name,
+        compile_options,
+        fuzz_folder_config,
+        fuzz_execution_config,
+    )
+}
+
+fn get_fuzzing_harnesses_in_package(
+    file_manager: &FileManager,
+    parsed_files: &ParsedFiles,
+    package: &Package,
+    fn_name: &FunctionNameMatch,
+    options: &CompileOptions,
+) -> Result<Vec<String>, CliError> {
+    let (mut context, crate_id) = prepare_package(file_manager, parsed_files, package);
+    check_crate_and_report_errors(&mut context, crate_id, options)?;
+
+    Ok(context
+        .get_all_fuzzing_harnesses_in_crate_matching(&crate_id, fn_name)
+        .into_iter()
+        .map(|(test_name, _)| test_name)
+        .collect())
+}
+
+fn display_fuzzing_report_and_store(
+    root_path: Option<PathBuf>,
+    fuzzing_failure_folder: Option<String>,
+    file_manager: &FileManager,
+    package: &Package,
+    compile_options: &CompileOptions,
+    fuzzing_report: &[(String, FuzzingRunStatus)],
+) -> Result<(), CliError> {
+    let writer = StandardStream::stderr(ColorChoice::Always);
+    let mut writer = writer.lock();
+    let fuzzing_failure_path =
+        fuzzing_failure_folder.map(PathBuf::from).unwrap_or(root_path.clone().unwrap_or_default());
+    if !fuzzing_failure_path.exists() {
+        std::fs::create_dir_all(&fuzzing_failure_path)
+            .expect("Failed to create fuzzing failure directory");
+    }
+
+    for (fuzzing_harness_name, test_status) in fuzzing_report {
+        write!(writer, "[").expect("Failed to write to stderr");
```

Comment:
Given how unlikely these are to fail, we could use `write!(...)?;` here and everywhere to reduce the noise and make the code clearer (the test_cmd code used to be like this but it was eventually changed).

Reply:
Maybe in a follow-up. It already uses cli-error, so I'd have to put these into a helper to convert